docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,151 @@
+Centralized CLI
+===============
+
+Context
+-------
+
+MetalK8s comes with a set of services to operate and monitor the K8s cluster.
+All operations that need to be performed by the Platform Administrator could be
+categorized as follow:
+
+- Cluster Resources Administration (Nodes, Volumes, Deployments, ...)
+- Cluster Administration (Install, Upgrade, Downgrade, Backup, Restore, ...)
+- Solution Administration (CRUD Environment, Import/Remove Solution, ...)
+- Cluster Service Administration (Configure Dex, Prometheus, Alert Manager,
+  ...)
+
+K8s provides the kubectl CLI, enabling all kind of interactions with all
+Kubernetes resources, through k8s apiserver, but its usage often requires to
+build verbose YAML files. Also it does not leverage everything MetalK8s exposes
+through the salt API.
+It is shipped as an independent package and can be deployed and run from
+anywhere, on any OS.
+
+Currently, MetalK8s provides other set of scripts or manual procedures, but
+those are located in various locations, their usage may vary and they are not
+developed using the same logic.
+
+This makes the CLI and associated documentation not super intuitive and it
+also makes the maintenance more expensive in the long term.
+
+The goal of the project is to provide MetalK8s administrator with an intuitive
+and easy to use set of tools in order to administrate and operate a finite set
+of functionalities.
+
+Because kubectl is already in place and is well known by Kubernetes
+administrators, it will be used as a standard to follow, as much as possible,
+for all other MetalK8s CLIs:
+
+- CLI provides an exhaustive help, per action, with relevant examples
+- CLI provides <action> help when the command is not valid
+- CLI is not interactive (except if password input is needed)
+- CLI should not require password input
+- CLI provides a dryrun mode for intrusive operations
+- CLI provides a verbose (or debug) mode
+- CLI implementation relies on secure APIs
+- CLI support action completion for easy discovery
+- CLI output is standardized and human readable by default
+- CLI output can be formatted in JSON or YAML
+
+When it is possible, it would make sense to leverage kubectl plugin
+
+Most functionalities are exposed through 2 distinct CLI:
+
+- kubectl: enriched with metalk8s plugin, to interact with both k8s apiserver
+  and salt API, and that can be executed from outside of the cluster.
+- metalk8sctl: a new CLI, exposing specific MetalK8s functionalities, that are
+  not interacting with k8s apiserver, and that must be executed on cluster node
```

Comment:
> that are not interacting with k8s apiserver

It's not true for me, some of them interact with apiserver just they must be run on the machine because they interact also with local machine

Reply:
ok, will do that in one other PR if you don't mind